OSDOCS-11208-networking: Bug batched 4.17 Networking bugs (#82473)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

[id="ocp-4-17-networking-bug-fixes_{context}"]
==== Networking

> * Previously, the SR-IOV Network Operator was listing the `SriovNetworkNodePolicies` resources in random order. This caused the `sriov-device-plugin` pod to enter a continuous restart loop. With this release, the SR-IOV Network Operator lists policies in a deterministic order so that
> the `sriov-device-plugin` pod does not enter a continuous restart loop. (link:https://issues.redhat.com/browse/OCPBUGS-36243[*OCPBUGS-36243*])

Reviewer: ```suggestion
* Previously, the SR-IOV Network Operator was listing the `SriovNetworkNodePolicies` resources in random order. This caused the `sriov-device-plugin` pod to enter a continuous restart loop. With this release, the SR-IOV Network Operator lists policies in a deterministic order so that the `sriov-device-plugin` pod does not enter a continuous restart loop. (link:https://issues.redhat.com/browse/OCPBUGS-36243[*OCPBUGS-36243*])
```

Author: :heavy_check_mark:

> * Previously,  an interface created inside a new pod would remain inactive and the Gratuitous Address Resolution Protocol (GARP) notification would be generated. The notification did not reach the cluster and ARP tables of other pods inside the cluster did not get updated with the MAC address of the new pod. This situation caused cluster traffic to stall until ARP table entries expired. With this release, a GARP notification is now sent after the interface inside a pod is active so that the GARP notification reaches the cluster. This means that surrounding pods can identify the new pod earlier than the previous behavior. (link:https://issues.redhat.com/browse/OCPBUGS-30549[*OCPBUGS-30549*])

Author: :heavy_check_mark:

> * Previously, enabling FIPS for a cluster causes SR-IOV device plugin pods to fail. With this release, SR-IOV device plugin pods have FIPS enabled so that when you enable FIPS for the cluster, the pods do not fail. (link:https://issues.redhat.com/browse/OCPBUGS-41131[*OCPBUGS-41131*])

Author: :heavy_check_mark:

> * Previously, a race condition was generated after rebooting an {product-title} node that used a performance profile with a small number of reserved CPUs. This occurred because Single Root I/O Virtualization (SR-IOV) virtual functions (VFs) shared the same MAC address and any pods that use the VFs would experience communication issues. With this release, an update to the SR-IOV Network Operator config daemon ensures that the Operator checks that no duplicate MAC addresses exist on VFs. (link:https://issues.redhat.com/browse/OCPBUGS-33137[*OCPBUGS-33137*])

Author: :heavy_check_mark:

* Previously, if you deleted the `sriovOperatorConfig` custom resource (CR), you could not create a new `sriovOperatorConfig` CR. With this release, the Single Root I/O Virtualization (SR-IOV) Network Operator removes validating webhooks when you delete the `sriovOperatorConfig` CR, so that you can create a new `sriovOperatorConfig` CR.   (link:https://issues.redhat.com/browse/OCPBUGS-37567[*OCPBUGS-37567*])

> * Previously, when you switched your cluster to use a different load balancer, the Ingress Operator did not remove the values in the `classicLoadBalancer` and `networkLoadBalancer` parameters in the `IngressController` custom resource (CR) status. This situation caused the status of the CR to report wrong information from the `classicLoadBalancer` and `networkLoadBalancer` parameters. With this release, after you switch your cluster to use a different load balancer, the Ingress Operator removes values from these parameters so that the CR represents a more exact and less confusing message status. (link:https://issues.redhat.com/browse/OCPBUGS-38646[*OCPBUGS-38646*])

Author: :heavy_check_mark:

> * Previously, all multicast packets did not reach their intended target nodes when a multicast sender and a multicast receiver existed on the same node. This happened because of an OVN-Kubernetes RPM package update. With this release, this regression is fixed in the OVN-Kubernetes RPM package, so that this issue no longer persists.(link:https://issues.redhat.com/browse/OCPBUGS-34778[*OCPBUGS-34778*])

Author: :heavy_check_mark:

> * Previously, when you created a `LoadBalancer` service for the Ingress Operator, a log message was generated that stated the change was not effective. This log message should only trigger for a change to an Infra custom resource. With this release, this log message is no longer generated when you create a `LoadBalancer` service for the Ingress Operator. (link:https://issues.redhat.com/browse/OCPBUGS-34413[*OCPBUGS-34413*])

Author: :heavy_check_mark:

> * Previously, for DNS names that had IP addresses with expired time-to-live (TTL) values, the `DNSNameResolver` controller that the Cluster DNS Operator manages would send DNS requests to CoreDNS pods. This caused the continuous generation of DNS requests and memory leaks issues for the pods. With this release, the `DNSNameResolver` controller waits until it receives the updated list of IP addresses and TTL values for a DNS name before the controller sends any more requests to the DNS name. The controller no longer generates erroneous requests and sends them to pods. CoreDNS pods can now respond to DNS requests in a timely manner and update the `DNSNameResolver` objects with the latest IP addresses and TTLs. (link:https://issues.redhat.com/browse/OCPBUGS-33750[*OCPBUGS-33750*])

Author: :heavy_check_mark:

> * Previously, when you used the `must-gather` tool, a Multus Container Network Interface (CNI) log file, `multus.log`, was stored in a node's file system. This situation caused the tool to generate unnecessary debug pods in a node. With this release, the Multus CNI no longer creates a `multus.log` file, but instead uses a CNI plugin pattern to inspect any logs for Multus DaemonSet pods in the `openshift-multus` namespace. (link:https://issues.redhat.com/browse/OCPBUGS-33959[*OCPBUGS-33959*])

Author: :heavy_check_mark:

* Previously, an alert for `OVNKubernetesNorthdInactive` would not fire in circumstances where it should fire. With this release, the issue is fixed so that the alert for `OVNKubernetesNorthdInactive` fires as expected. (link:https://issues.redhat.com/browse/OCPBUGS-33758[*OCPBUGS-33758*])

> * Previously, a missing route for the Kubernetes-OVN masquerade address in all pods in pods where the default route has been customized would cause each pod to be unable to connect to itself through a service for which it acts as backend. With this release, the missing route for Kubernetes-OVN masquerade address is added to pods so that the issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-36865[*OCPBUGS-36865*])

Author: :heavy_check_mark:

> * Previously, the `iptables-alerter` pod did not handle errors from `crictl`, which could cause the pod to incorrectly log events from `host-network` pods or cause pod restarts. With this release, the errors are handled correctly so that these issues no longer persist. (link:https://issues.redhat.com/browse/OCPBUGS-37713[*OCPBUGS-37713*])

Author: :heavy_check_mark:

* Previously, if you created a hosted cluster by using a proxy for the purposes of making the cluster reach a control plane from a compute node, the compute node would be unavailable to the cluster. With this release, the proxy settings are updated for the node so that the node can use a proxy to successfully communicate with the control plane. (link:https://issues.redhat.com/browse/OCPBUGS-37786[*OCPBUGS-37786*])

> * Previously,  when a cluster failed to install on an on-premise platform with a configured load balancer, the `LoadBalancer` service's `LoadBalancerReady` condition received the `SyncLoadBalancerFailed` status. The status would trigger the generation of the `The kube-controller-manager logs might contain more details.` message. This message is wrong because the logs are stored in the `cloud-controller-manager` namespace of a project. With this release,  the `SyncLoadBalancerFailed` status now communicates the correct message: `The cloud-controller-manager logs may contain more details`. (link:https://issues.redhat.com/browse/OCPBUGS-31664[*OCPBUGS-31664*])

Author: :heavy_check_mark:

> * Previously, you could not control log levels for the internal component that selects IP addresses for cluster nodes. With this release, you can now enable debug log levels so that you can either increase or decrease log levels on-demand. To adjust log levels, you must create a config map manifest file with a configuration analogous to the following:

Reviewer: ```suggestion
* Previously, you could not control log levels for the internal component that selects IP addresses for cluster nodes. With this release, you can now enable debug log levels so that you can either increase or decrease log levels on-demand. To adjust log levels, you must create a config map manifest file with a configuration similar to the following:
```

Author: :heavy_check_mark:

+
[source,yaml]
----
apiVersion: v1
data:
  enable-nodeip-debug: "true"
kind: ConfigMap
metadata:
  name: logging
  namespace: openshift-vsphere-infra
# ...
----
+
(link:https://issues.redhat.com/browse/OCPBUGS-32348[*OCPBUGS-32348*])

> * Previously, the Ingress Operator could not successfully update the canary route because the Operator did not have permission to update `spec.host` or `spec.subdomain` on an existing route. With this release, the required permission is added to the cluster role for the Operator’s service account and the Ingress Operator can update the canary route. (link:https://issues.redhat.com/browse/OCPBUGS-36465[*OCPBUGS-36465*])

Reviewer: ```suggestion
* Previously, the Ingress Operator could not successfully update the canary route because the Operator did not have permission to update `spec.host` or `spec.subdomain` fields on an existing route. With this release, the required permission is added to the cluster role for the Operator’s service account and the Ingress Operator can update the canary route. (link:https://issues.redhat.com/browse/OCPBUGS-36465[*OCPBUGS-36465*])
```

Author: :heavy_check_mark:

* Previously, administrator privileges were required to run some networking containers, such as Keepalived, on supported on-premise platforms. With this release, these containers no longer require administrator privileges to run them on supported on-premise platforms. (link:https://issues.redhat.com/browse/OCPBUGS-36175[*OCPBUGS-36175*])

> * Previously, if your `NodeNetworkConfigurationPolicy` (NNCP) custom resource (CR) is set to use the default spanning tree protocol (STP) implementation, the CR YAML file would show `stp.enabled: true`, but the {product-title} web console deselected the STP checkbox. This situation is misleading. With this release, the web console only deselects the STEP checkbox after you define `stp.enabled: false` in the NNCP CR YAML file.

Reviewer: ISG nits:
```suggestion
* Previously, if your `NodeNetworkConfigurationPolicy` (NNCP) custom resource (CR) is set to use the default spanning tree protocol (STP) implementation, the CR configuration file would show `stp.enabled: true`, but the {product-title} web console cleared the STP check box. With this release, the web console only clears the STEP check box after you define `stp.enabled: false` in the NNCP CR YAML file.
```

Author: :heavy_check_mark:

(link:https://issues.redhat.com/browse/OCPBUGS-36238[*OCPBUGS-36238*])

> * Previously, the ingress controller status was incorrectly displayed as `Degraded=False` because of a migration time issue with the `CanaryRepetitiveFailures` condition. With this release, the ingress controller is correctly marked as `Degraded=True` for the appropriate length of time that the `CanaryRepetitiveFailures` condition exists. (link:https://issues.redhat.com/browse/OCPBUGS-39220[*OCPBUGS-39220*])

Reviewer: ```suggestion
* Previously, the Ingress Controller status incorrectly displayed as `Degraded=False` because of a migration time issue with the `CanaryRepetitiveFailures` condition. With this release, the Ingress Controller status is correctly marked as `Degraded=True` for the appropriate length of time that the `CanaryRepetitiveFailures` condition exists. (link:https://issues.redhat.com/browse/OCPBUGS-39220[*OCPBUGS-39220*])
```

Author: :heavy_check_mark:

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

> [id="ocp-4-17-known-issues"]

Reviewer: ```suggestion
[id="ocp-4-17-known-issues_{context}"]
```

Author: :heavy_check_mark:
